cstar/base/component.py

```diff
@@ -46,32 +55,26 @@ def __init__(self, **kwargs: Any):
         base_model: BaseModel
             An object pointing to the unmodified source code of a model handling an individual
             aspect of the simulation such as biogeochemistry or ocean circulation
-        additional_code: AdditionalCode or list of AdditionalCodes
+        additional_code: AdditionalCode (Optional, default None)
             Additional code contributing to a unique instance of a base model,
             e.g. namelists, source modifications, etc.
-        input_datasets: list of InputDatasets
+        input_datasets: list of InputDatasets (Optional, default [])
             Any spatiotemporal data needed to run this instance of the base model
             e.g. initial conditions, surface forcing, etc.
+        discretization: Discretization (Optional, default None)
+            Any information related to the discretization of this Component
+            e.g. time step, number of vertical levels, etc.
+        
 
         Returns:
         --------
         Component:
             An intialized Component object
         """
-
-        # TODO: do Type checking here
-        if "base_model" not in kwargs or not isinstance(
-            kwargs["base_model"], BaseModel
-        ):
-            raise ValueError(
-                "base_model must be provided and must be an instance of BaseModel"
-            )
```

Comment:
You seem to have lost these instance checks. We could put them in property setters, but only if properties are supposed to be settable. We could autogenerate the `isinstance` checks by using a `dataclass`, or we could just write them out here.

Reply:
reinstated this in b69800f70ca3a583ffdde906d663d847df97404b